python/lib/core/dmod/core/meta_data.py

```diff
@@ -528,6 +659,35 @@ def contains(self, other: 'DiscreteRestriction') -> bool:
                     return False
         return True
 
+    def extend(self, other: 'DiscreteRestriction') -> 'DiscreteRestriction':
+        """
+        Produce another instance made by extending this instance with the other, assuming compatibility.
+
+        Parameters
+        ----------
+        other: DiscreteRestriction
+            Another restriction object.
+
+        Returns
+        -------
+        DiscreteRestriction
+            A new instance representing the combined restriction defined by this and the other instance.
+
+        Raises
+        ------
+        DmodRuntimeError
+            Raised if the two instances are not compatible as testable via :method:`can_extend`
+
+        See Also
+        --------
+        can_extend
+        """
+        if not self.can_extend_with(other):
+            raise DmodRuntimeError(f"Attempting to extend incompatible {self.__class__.__name__} objects")
+        if other.is_all_possible_values:
+            return DiscreteRestriction(**other.dict())
+        return DiscreteRestriction(variable=self.variable, values=self.values + other.values)
```

Comment:
again, this seems to be much more of a `combine` or `merge` action than an extension.